### ISSUE TITLE
react-loadable is out of date

### ISSUE THREAD

**Issue by woobianca** [curioswitch/curiostack#488]
Lifecycle methods for this library need to be updated, which affects page rendering.

Found this issue when trying to redirect to a page after checking if state changed, but it never redirects.

```
Warning: componentWillMount has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.

* Move code with side effects to componentDidMount, and set initial state in the constructor.
* Rename componentWillMount to UNSAFE_componentWillMount to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: LoadableComponent
```

Directly rendering my component, rather than using the LoadableComponent fixes my issue in the short term, but might need to migrate over to another library since react-loadable seems abandoned.
@danada

**Comment by danada** [comment]
Instead of migrating to a new library, there might be a way to natively (`React.suspense`) cover all the functionality of react-loadable, however server-side rendering isn't supported yet 😿 . 

React's official docs recommend https://github.com/smooth-code/loadable-components in the meantime.

**Comment by chokoswitch** [comment]
Yeah I'm hoping to skip a migration to loadable-component and then suspense by waiting for SSR support in suspense. I'm pretty sure this is supposed to happen before React 17, so until then let's just live with the warning.

The warning itself seems like a red herring since the API rename doesn't actually change functionality or page rendering. So far I haven't come across brokenness related to react-loadable, but can you describe better what your code flow looks like? Is it only certain redirects or all of them that don't work?

**Comment by danada** [comment]
Just an update on this: I think a lot of these issues are stemming from the changes addressed in #492 , so once released we could probably confirm this isn't causing issues and close this up :smile_cat:

**Comment by woobianca** [comment]
#492 fixes my redirect flow.

Will close this issue since the outdated `react-loadable` only creates console warnings but is not causing any real problems.
